### PR TITLE
Add enable pin for stepper

### PR DESCRIPTION
**Pull request by dtex**
Will update the wiki when this lands.

### CODE REVIEW

**Comment by dtex** [comment]
Has a bug. Will fix and resubmit.

File: lib/stepper.js [hunk 1]
```diff
@@ -157,13 +158,17 @@ function Stepper(opts) {
...
     if (opts.type === Stepper.TYPE.DRIVER) {
-      this.pins = {
-        step: this.pins[0],
-        dir: this.pins[1]
-      };
+      this.pins.step = pinArray[0];
+      this.pins.dir = pinArray[1];
     } else {
-      this.pins = new MotorPins(this.pins);
+      this.pins = new MotorPins(pinArray);
```
Comment (rwaldron) [inline lib/stepper.js]: Blah... `Stepper` is such a mess in the way it handles `this.pins` :(

Comment (dtex) [inline lib/stepper.js]: Ya, that's what made me close my PR. I missed something in the convolution.

**Comment by dtex** [comment]
Okay, so this branch works fine for the HRC robot so my urgency has abated. If I go ahead and refactor this class I am going to use this [boilerplate for Johnny-Five classes](https://gist.github.com/dtex/1ef97480209efe621cf9). Would you like to suggest any changes before I do?

**Comment by rwaldron** [comment]
I like all of those comments :)


Using `this.opts.anything` has to go (should've never happened in motor.js, which it only appears there iirc).


Change https://gist.github.com/dtex/1ef97480209efe621cf9#file-classboilerplate-js-L91 to: 

```js
  Board.Component.call(
    // opts gets normalized to opts, not `this.opts`
    this, opts = Board.Options(opts)
  );
```

Change https://gist.github.com/dtex/1ef97480209efe621cf9#file-classboilerplate-js-L97-L103 to: 

```js
  // If this class has multiple devices we need to figure out which one to use
  if (typeof opts.device === "undefined") {
    // Derive the device type if not explicitly passed
  }

  // Allow users to pass in custom device types
  device = typeof opts.device === "string" ?
    Devices[opts.device] : opts.device;
```

Change https://gist.github.com/dtex/1ef97480209efe621cf9#file-classboilerplate-js-L112-L118 to:


```js
  // Then the controller decorates the instance
  var controller = null;

  if (opts.controller && typeof opts.controller === "string") {
    controller = Controllers[opts.controller.toUpperCase()];
  } else {
    controller = opts.controller;
  }

  if (controller == null) {
    controller = Controllers.ANALOG;
  }

  Board.Controller.call(this, controller, opts);
```


All occurrences of `this.opts.someValue` => `this.someValue`

**Comment by rwaldron** [comment]
So.. this: 

```js
  if (controller == null) {
    controller = Controllers.ANALOG;
  }
```

You should choose an intuitive controller as a default (or if you decide "no default" it needs to throw): 

```js
  if (controller == null) {
    throw new Error("Stepper expects a valid controller");
  }
```

**Pull request closed without merging**